Lib/inspect.py

```diff
@@ -395,9 +395,9 @@ def _has_code_flag(f, flag):
     while ismethod(f):
         f = f.__func__
     f = functools._unwrap_partial(f)
-    if not isfunction(f):
-        return False
-    return bool(f.__code__.co_flags & flag)
+    if isfunction(f) or _signature_is_functionlike(f):
+        return bool(f.__code__.co_flags & flag)
```

Comment:
Can you try to preserve current coding style? Return early with False if a condition is false? Pseudo-code:

```
if not isfunction(f): return False
if not _signature_is_functionlike(f): return False
return bool(f.__code__.co_flags & flag)
```

Reply:
done